src/contracts/proxies/Vault721.sol

```diff
@@ -112,15 +113,15 @@ contract Vault721 is ERC721EnumerableUpgradeable {
    * @dev allows msg.sender without an ODProxy to deploy a new ODProxy
    */
   function build() external returns (address payable _proxy) {
-    if (!_isNotProxy(msg.sender)) revert ProxyAlreadyExist();
+    if (!_buildProxy(msg.sender)) revert ProxyAlreadyExist();
```

Comment:
Is the check still needed here, since its already performed downstream in `_build`?

Reply:
this is to check if address does not own a proxy. buildProxy? yes or no

then in the _build function, there is a check to make the address is not a proxy itself, so the checks are different